include/nbl/asset/ICPUPolygonGeometry.h

```diff
@@ -0,0 +1,176 @@
+
+        // Needs to be hidden because ICPU base class shall check mutability
+        inline bool setIndexView(SDataView&& view)
+        {
+            if (isMutable())
+                return base_t::setIndexView(std::move(view));
+            return false;
+        }
+
+        //
+        inline bool setVerticesForFirst(const uint16_t count)
+        {
+            if (isMutable())
+            {
+                m_verticesForFirst = count;
+                return true;
+            }
+            return false;
+        }
+
+        //
+        inline bool setVerticesPerSupplementary(const uint16_t count)
+        {
+            if (isMutable())
+            {
+                m_verticesPerSupplementary = count;
+                return true;
+            }
+            return false;
+        }
+
+        //
+        inline void setNormalView(SDataView&& view)
+        {
+            if (isMutable())
+            {
+                m_normalView = std::move(view);
+                return true;
+            }
+            return false;
+        }
+
+        //
+        inline bool setJointCount(const uint32_t count)
+        {
+            if (isMutable())
+            {
+                m_jointCount = count;
+                return true;
+            }
+            return false;
+        }
+
+        //
+        inline const core::vector<SJointWeight>* getJointWeightViews()
+        {
+            if (isMutable())
+                return m_jointWeightViews;
+            return nullptr;
+        }
+
+        //
+        inline const core::vector<SJointWeight>* getAuxAttributeViews()
+        {
+            if (isMutable())
+                return m_auxAttributeViews;
+            return nullptr;
+        }
+
+    protected:
+        //
+        inline void visitDependents(std::function<bool(const IAsset*)>& visit) const //override
+        {
+            if (!visit(m_positionView.src.buffer.get())) return;
+            if (!visit(m_jointOBBView.src.buffer.get())) return;
+            if (!visit(m_indexView.src.buffer.get())) return;
+            for (const auto& pair : m_jointWeightViews)
+            {
+                if (!visit(pair.indices.src.buffer.get())) return;
+                if (!visit(pair.weights.src.buffer.get())) return;
+            }
+            for (const auto& view : m_auxAttributeViews)
+                if (!visit(view.src.buffer.get())) return;
+            if (!visit(m_normalView.src.buffer.get())) return;
+        }
+        // TODO: remove after https://github.com/Devsh-Graphics-Programming/Nabla/pull/871 merge
+        inline IAsset* getDependant_impl(const size_t ix) override
+        {
+           const IAsset* retval = nullptr;
+           size_t current = 0;
+           visitDependents([&current](const IAsset* dep)->bool
+               {
+                   retval = dep;
+                   return ix<current++;
+               }
+           );
+           return const_cast<IAsset*>(retval);
+        }
```

Comment:
@kevyuu this is what I had in mind for the visitor pattern & callback https://github.com/Devsh-Graphics-Programming/Nabla/pull/871